src/Prometheus.Client.HttpClient/MessageHandlers/HttpClientMessageHandlerBase.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Net.Http;
+
+namespace Prometheus.Client.HttpClient.MessageHandlers
+{
+    public abstract class HttpClientMessageHandlerBase<TMetricFamily, TMetric> : DelegatingHandler
+        where TMetricFamily: IMetricFamily<TMetric, ValueTuple<string, string, string, string>>
+        where TMetric : IMetric
+    {
+        private readonly string _clientName;
+        private readonly TMetricFamily _metric;
+        protected readonly IMetricFactory MetricFactory;
+        protected abstract TMetricFamily CreateMetricInstance();
+        protected HttpClientMessageHandlerBase(IMetricFactory metricFactory, string clientName)
+        {
+            _clientName = clientName;
+            MetricFactory = metricFactory;
+            _metric = CreateMetricInstance();
```

Comment:
Warning: https://www.jetbrains.com/help/resharper/VirtualMemberCallInConstructor.html

Reply:
@phnx47 
I think for now we can ignore this since we don't have any implementation on derived constructors as of now.